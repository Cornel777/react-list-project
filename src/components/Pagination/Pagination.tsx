import { Button } from "@mui/material";
import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import { FunctionComponent } from "react";
import LoadingDots from "../LoadingDots/LoadingDots";
import CurrentPageContainer from "./styled/CurrentPageContainer";

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  hasNextPage: boolean;
  isLoading: boolean;
  isPlaceholderData: boolean;
  onPageChange: (page: number) => void;
}

const Pagination: FunctionComponent<PaginationProps> = ({ 
  currentPage, 
  totalPages, 
  hasNextPage,
  isLoading,
  isPlaceholderData, 
  onPageChange 
}) => {
  return (
    <>
      <CurrentPageContainer>Current Page: {currentPage}/{totalPages}</CurrentPageContainer>
      <Button
        sx={{margin: '0 auto'}}
        variant='contained'
        onClick={() => onPageChange(Math.max(currentPage - 1, 0))}
        disabled={currentPage === 1}
      >
        <ArrowBackIosIcon/> Previous Page
      </Button>{' '}
      <Button
        sx={{margin: '0 auto'}}
        variant='contained'
        onClick={() => onPageChange(currentPage + 1)}
        disabled={ isPlaceholderData &&  !hasNextPage}
      >
        Next Page <ArrowForwardIosIcon />
      </Button>
      {isLoading ? <LoadingDots/>: null}
    </>
  );
}

export default Pagination;