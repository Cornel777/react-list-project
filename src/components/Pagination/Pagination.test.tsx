import { describe, expect, it, vi } from "vitest";
import { render} from "@testing-library/react";
import { fireEvent } from '@testing-library/react';
import Pagination from "./Pagination";

describe('Pagination', () => {
  const defaultProps = {
    currentPage: 1,
    totalPages: 3,
    hasNextPage: true,
    isLoading: false,
    isPlaceholderData: false,
    onPageChange: vi.fn(),
  };


  it('should render the pagination component', () => {
      const { container } = render(<Pagination {...defaultProps} />);
      expect(container).toMatchSnapshot();
  });

  it('should render the pagination component with the current page', () => {
    const { getByText } = render(<Pagination {...defaultProps} />);
    expect(getByText('Current Page: 1/3')).toBeInTheDocument();
  });

  it('should render the pagination component with both buttons', () => {
    const { getByText } = render(<Pagination {...defaultProps} />);
    expect(getByText('Previous Page')).toBeInTheDocument();
    expect(getByText('Next Page')).toBeInTheDocument();
  });

  it('disables the previous page button when currentPage is 1', () => {
    const { getByText } = render(<Pagination {...defaultProps} currentPage={1} />);
    expect(getByText('Previous Page')).toBeDisabled();
  });

  it('disables the next page button when isPlaceholderData is true and hasNextPage is false', () => {
    const { getByText } = render(<Pagination {...defaultProps} isPlaceholderData={true} hasNextPage={false} />);
    expect(getByText('Next Page')).toBeDisabled();
  });

  it('calls onPageChange with the previous page number when the previous page button is clicked', () => {
    const {getByText} = render(<Pagination {...defaultProps} currentPage={2} />);
    fireEvent.click(getByText('Previous Page'));
    expect(defaultProps.onPageChange).toHaveBeenCalledWith(1);
  });
});