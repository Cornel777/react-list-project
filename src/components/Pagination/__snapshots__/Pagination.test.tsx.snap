// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should render the pagination component 1`] = `
<div>
  <div
    class="sc-gyycJP"
  >
    Current Page: 
    1
    /
    3
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1l0eyhc-MuiButtonBase-root-MuiButton-root"
    disabled=""
    id=":r0:"
    tabindex="-1"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="ArrowBackIosIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
      />
    </svg>
     Previous Page
  </button>
   
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1l0eyhc-MuiButtonBase-root-MuiButton-root"
    id=":r1:"
    tabindex="0"
    type="button"
  >
    Next Page 
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
      data-testid="ArrowForwardIosIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
      />
    </svg>
  </button>
</div>
`;
